src/datatrove/pipeline/readers/videotriplet.py

```diff
@@ -0,0 +1,170 @@
+import json
+import warnings
+import os
+from typing import Callable, List, Dict
+from datatrove.io import DataFileLike, DataFolderLike, get_datafolder, download_file
+from datatrove.pipeline.readers.base import BaseDiskReader
+from datatrove.data import Document, Media, MediaType
+from datatrove.data import Document, DocumentsPipeline
+
+class VideoTripletReader(BaseDiskReader):
+    """Read triplets of video, metadata, and optional caption files."""
+
+    name = "🎥 Video Triplet Reader"
+
+    def __init__(
+        self,
+        data_folder: DataFolderLike,
+        paths_file: DataFileLike | None = None,
+        metadata_origin: str | None = None,
+        limit: int = -1,
+        skip: int = 0,
+        file_progress: bool = False,
+        doc_progress: bool = False,
+        adapter: Callable = None,
+        text_key: str = "text",
+        id_key: str = "id",
+        default_metadata: dict = None,
+        recursive: bool = True,
+        glob_pattern: str | None = None,
+        local_cache_dir = "/tmp/local_video_cache"
+    ):
+        self.data_folder = get_datafolder(data_folder)
+        self.paths_file = paths_file
+        self.metadata_origin = metadata_origin
+        self.local_cache_dir = local_cache_dir  
+        os.makedirs(self.local_cache_dir, exist_ok=True)
+        super().__init__(
+            data_folder,
+            paths_file,
+            limit,
+            skip,
+            file_progress,
+            doc_progress,
+            adapter,
+            text_key,
+            id_key,
+            default_metadata,
+            recursive,
+            glob_pattern,
+        )
+
+
+    def run(self, data: DocumentsPipeline, rank: int = 0, world_size: int = 1) -> DocumentsPipeline:
+        """Overrides the base run method to handle triplet statistics correctly."""
+        triplet_count = 0
+
```

Comment:
This allows stacking multiple readers together (when you want to load data from multiple sources for example)
```suggestion
        if data:
            yield from data
```